sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala

```diff
@@ -129,7 +129,81 @@ case class GpuLocalLimitExec(limit: Int, child: SparkPlan) extends GpuBaseLimitE
  * Take the first `limit` elements of the child's single output partition.
  */
 case class GpuGlobalLimitExec(limit: Int, child: SparkPlan, offset: Int) extends GpuBaseLimitExec {
+  assert(limit >= 0 || (limit == -1 && offset > 0))
   override def requiredChildDistribution: List[Distribution] = AllTuples :: Nil
+
+  override def doExecuteColumnar(): RDD[ColumnarBatch]  = {
+    val opTime = gpuLongMetric(OP_TIME)
+    val childRdd = child.executeColumnar()
+    childRdd.mapPartitions {
+      iter => new Iterator[ColumnarBatch] {
+        private var remainingLimit = limit - offset
+        private var remainingOffset = offset
+
+        override def hasNext: Boolean = remainingLimit > 0 && iter.hasNext
+
+        override def next(): ColumnarBatch = {
+          val batch = iter.next()
+
+          // If remainingOffset >= numRows, then we should skip this batch, and return an empty one
```

Comment:
We should not return an empty one, but get next batch from children, until skipping `offset` rows in total.